File: src/assets-source/courses/db/lab02/solution/src/main/java/com/github/szgabsz91/iitszg/db/lab02/exercise01/model/Car.java
package com.github.szgabsz91.iitszg.db.lab02.exercise01.model;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "Cars")
@NoArgsConstructor
@EqualsAndHashCode(exclude = "owner")
@ToString(exclude = "owner")
public class Car {

    @Id
    @Getter
    @Setter
    private String licensePlateNumber;

    @Column(nullable = false)
    @Getter
    @Setter
    private String manufacturer;

    @Column(nullable = false)
    @Getter
    @Setter
    private String model;

    @Column(nullable = false)
    @Getter
    @Setter
    private double price;

    @ManyToOne(optional = false)
    @Getter
    @Setter
    private Person owner;

    public Car(String licensePlateNumber, String manufacturer, String model, double price, Person owner) {
        this.licensePlateNumber = licensePlateNumber;
        this.manufacturer = manufacturer;
        this.model = model;
        this.price = price;
        this.owner = owner;

        if (!this.owner.getCars().contains(this)) {
            this.owner.getCars().add(this);
        }
    }

}

File: src/assets-source/courses/db/lab03/solution/Model/Car.cs
namespace Lab03.Model
{
    public class Car
    {
        public string LicensePlateNumber { get; set; }

        public string Manufacturer { get; set; }

        public string Model { get; set; }

        public decimal Price { get; set; }

        public string Owner { get; set; }

        public override string ToString()
        {
            return string.Format("Car[LicensePlateNumber = {0}, Manufacturer = {1}, Model = {2}, Price = {3}, Owner = {4}]", LicensePlateNumber, Manufacturer, Model, Price, Owner);
        }
    }
}
